move_base/src/move_base.cpp

```diff
@@ -97,11 +97,14 @@ namespace move_base {
     current_goal_pub_ = private_nh.advertise<geometry_msgs::PoseStamped>("current_goal", 0 );
     ros::NodeHandle action_nh("move_base");
     action_goal_pub_ = action_nh.advertise<move_base_msgs::MoveBaseActionGoal>("goal", 1);
+    action_goal_sub_ = action_nh.subscribe<move_base_msgs::MoveBaseActionGoal>("goal", 1, boost::bind(&MoveBase::actionGoalCB, this, _1));
```

Comment:
It's interesting that you send and receive them yourself. I also feel like I could have made a few calls internally.